src/category_theory/monoidal/free/basic.lean

```diff
@@ -0,0 +1,237 @@
+| of : C → free_monoidal_category
+| unit : free_monoidal_category
+| tensor : free_monoidal_category → free_monoidal_category → free_monoidal_category
+
+end
+
+local notation `F` := free_monoidal_category
+
+namespace free_monoidal_category
+
+/-- Formal compositions and tensor products of identities, unitors and associators. The morphisms
+    of the free monoidal category are obtained as a quotient of these formal morphisms by the
+    relations defining a monoidal category. -/
+@[nolint has_inhabited_instance]
+inductive hom : F C → F C → Type u
+| id (X) : hom X X
+| α_hom (X Y Z : F C) : hom ((X.tensor Y).tensor Z) (X.tensor (Y.tensor Z))
+| α_inv (X Y Z : F C) : hom (X.tensor (Y.tensor Z)) ((X.tensor Y).tensor Z)
+| l_hom (X) : hom (unit.tensor X) X
+| l_inv (X) : hom X (unit.tensor X)
+| ρ_hom (X : F C) : hom (X.tensor unit) X
+| ρ_inv (X : F C) : hom X (X.tensor unit)
+| comp {X Y Z} (f : hom X Y) (g : hom Y Z) : hom X Z
+| tensor {W X Y Z} (f : hom W Y) (g : hom X Z) : hom (W.tensor X) (Y.tensor Z)
+
+infixr ` ⟶ᵐ `:10 := hom
+
+/-- The morphisms of the free monoidal category satisfy 21 relations ensuring that the resulting
+    category is in fact a category and that it is monoidal. -/
+inductive hom_equiv : Π {X Y : F C}, (X ⟶ᵐ Y) → (X ⟶ᵐ Y) → Prop
+| refl {X Y} (f : X ⟶ᵐ Y) : hom_equiv f f
+| symm {X Y} (f g : X ⟶ᵐ Y) : hom_equiv f g → hom_equiv g f
+| trans {X Y} {f g h : X ⟶ᵐ Y} : hom_equiv f g → hom_equiv g h → hom_equiv f h
+| comp {X Y Z} {f f' : X ⟶ᵐ Y} {g g' : Y ⟶ᵐ Z} :
+    hom_equiv f f' → hom_equiv g g' → hom_equiv (f.comp g) (f'.comp g')
+| tensor {W X Y Z} {f f' : W ⟶ᵐ X} {g g' : Y ⟶ᵐ Z} :
+    hom_equiv f f' → hom_equiv g g' → hom_equiv (f.tensor g) (f'.tensor g')
+| comp_id {X Y} (f : X ⟶ᵐ Y) : hom_equiv (f.comp (hom.id _)) f
+| id_comp {X Y} (f : X ⟶ᵐ Y) : hom_equiv ((hom.id _).comp f) f
+| assoc {X Y U V : F C} (f : X ⟶ᵐ U) (g : U ⟶ᵐ V) (h : V ⟶ᵐ Y) :
+    hom_equiv ((f.comp g).comp h) (f.comp (g.comp h))
+| tensor_id {X Y} : hom_equiv ((hom.id X).tensor (hom.id Y)) (hom.id _)
+| tensor_comp {X₁ Y₁ Z₁ X₂ Y₂ Z₂ : F C}
+    (f₁ : X₁ ⟶ᵐ Y₁) (f₂ : X₂ ⟶ᵐ Y₂) (g₁ : Y₁ ⟶ᵐ Z₁) (g₂ : Y₂ ⟶ᵐ Z₂) :
+    hom_equiv ((f₁.comp g₁).tensor (f₂.comp g₂)) ((f₁.tensor f₂).comp (g₁.tensor g₂))
+| α_hom_inv {X Y Z} : hom_equiv ((hom.α_hom X Y Z).comp (hom.α_inv X Y Z)) (hom.id _)
+| α_inv_hom {X Y Z} : hom_equiv ((hom.α_inv X Y Z).comp (hom.α_hom X Y Z)) (hom.id _)
+| associator_naturality {X₁ X₂ X₃ Y₁ Y₂ Y₃} (f₁ : X₁ ⟶ᵐ Y₁) (f₂ : X₂ ⟶ᵐ Y₂) (f₃ : X₃ ⟶ᵐ Y₃) :
+    hom_equiv (((f₁.tensor f₂).tensor f₃).comp (hom.α_hom Y₁ Y₂ Y₃))
+      ((hom.α_hom X₁ X₂ X₃).comp (f₁.tensor (f₂.tensor f₃)))
+| ρ_hom_inv {X} : hom_equiv ((hom.ρ_hom X).comp (hom.ρ_inv X)) (hom.id _)
+| ρ_inv_hom {X} : hom_equiv ((hom.ρ_inv X).comp (hom.ρ_hom X)) (hom.id _)
+| ρ_naturality {X Y} (f : X ⟶ᵐ Y) : hom_equiv
+    ((f.tensor (hom.id unit)).comp (hom.ρ_hom Y)) ((hom.ρ_hom X).comp f)
+| l_hom_inv {X} : hom_equiv ((hom.l_hom X).comp (hom.l_inv X)) (hom.id _)
```

Comment:
Is there a reason you use `l` instead of `\lambda`?

Reply:
@gebner, is this a restriction that could easily be lifted?

Reply:
It's not really a restriction per se, `λ_hom_inv` just means something different, namely a lambda with an argument called `_hom_inv`. You can use French quotes though:
```lean
inductive Λ | «λ»
```